tor_core/helpers.py

```diff
@@ -252,6 +255,30 @@ def handle_rate_limit(exc):
     time.sleep(delay + 1)
 
 
+def signal_handler(signal, frame):
+    """
+    This is the SIGINT handler that allows us to intercept CTRL+C.
+    When this is triggered, it will wait until the primary loop ends
+    the current iteration before ending. Press CTRL+C twice to kill
+    immediately.
+
+    :param signal: Unused.
+    :param frame: Unused.
+    :return: None.
+    """
+    global running
+
+    if not running:
```

Comment:
I guess that’s one way of killing it on the second invocation. I figured we would have replaced the sigint handler in this handler such that it throws a KeyboardInterrupt exception again.

Doing it like this appears to function alright so I have no qualms. If there’s a more graceful way of handling signals we’ll iterate on it later. :+1: